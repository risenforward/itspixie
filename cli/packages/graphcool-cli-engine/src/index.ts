export { Cluster } from './Cluster'

export { readDefinition } from './GraphcoolDefinition/yaml'
export {
  GraphcoolDefinitionClass,
} from './GraphcoolDefinition/GraphcoolDefinition'

export { EnvDoesntExistError } from './errors/EnvDoesntExistError'

export { Config } from './Config'
export { Command } from './Command'
export { Environment } from './Environment'
export { CLI, run } from './CLI'
export { Output } from './Output'
export {
  RunOptions,
  ProjectDefinition,
  ProjectInfo,
  GraphcoolModule,
  RemoteProject,
  APIError,
  AuthServer,
  AuthTrigger,
  CheckAuth,
  MigrateProjectPayload,
  MigrationActionType,
  MigrationErrorMessage,
  MigrationResult,
  Project,
  Region,
  SchemaInfo,
  PAT,
  AuthenticateCustomerPayload,
  ExternalFiles,
  ExternalFile,
} from './types/common'

export { ClusterConfig, Clusters, RC } from './types/rc'

export { flags, Flag, Arg, Flags } from './Flags'

export { mockDefinition, mockEnv } from './mock'

export { Client } from './Client/Client'
