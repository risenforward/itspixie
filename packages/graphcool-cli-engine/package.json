{
  "name": "graphcool-cli-engine",
  "version": "0.0.6",
  "main": "dist/index.js",
  "files": [
    "dist"
  ],
  "typings": "dist/index.d.ts",
  "author": "Tim Suchanek <tim@graph.cool>",
  "license": "MIT",
  "devDependencies": {
    "@types/jest": "^20.0.8",
    "@types/node": "^8.0.22",
    "husky": "^0.14.3",
    "jest": "^21.0.1",
    "lint-staged": "^4.0.3",
    "nock": "^9.0.14",
    "prettier": "^1.5.3",
    "rimraf": "^2.6.1",
    "source-map-support": "^0.4.15",
    "ts-jest": "^21.0.0",
    "tslint": "^5.6.0",
    "tslint-config-prettier": "^1.3.0",
    "tslint-eslint-rules": "^4.1.1",
    "typescript": "^2.4.2"
  },
  "scripts": {
    "build": "rimraf dist && tsc -d",
    "lint": "tslint src/**/*.ts",
    "precommit": "lint-staged",
    "prettier": "prettier --single-quote --no-semi --trailing-comma all --write '*.ts' 'src/**/*.ts'",
    "prepublishOnly": "npm run lint && npm run build",
    "test": "jest && yarn lint"
  },
  "lint-staged": {
    "*.ts": [
      "prettier --single-quote --no-semi --trailing-comma all --write",
      "tslint",
      "git add"
    ]
  },
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json"
    ],
    "rootDir": "./src",
    "transform": {
      "^.+\\.(ts|tsx)$": "../../../node_modules/ts-jest/preprocessor.js"
    },
    "testMatch": [
      "**/*.test.(ts|js)"
    ],
    "setupFiles": [
      "../test/init.js"
    ]
  },
  "dependencies": {
    "@heroku/linewrap": "^1.0.0",
    "ansi-escapes": "^3.0.0",
    "chalk": "^2.1.0",
    "debug": "^3.0.1",
    "find-up": "^2.1.0",
    "fs-extra": "^4.0.1",
    "klaw-sync": "^3.0.0",
    "lodash.ary": "^4.1.1",
    "lodash.defaults": "^4.2.0",
    "lodash.flatten": "^4.4.0",
    "lodash.get": "^4.4.2",
    "lodash.identity": "^3.0.0",
    "lodash.keys": "^4.2.0",
    "lodash.maxby": "4.x",
    "lodash.merge": "4.x",
    "lodash.partial": "^4.2.1",
    "lodash.property": "^4.4.2",
    "lodash.result": "^4.5.2",
    "lodash.uniqby": "^4.7.0",
    "rwlockfile": "^1.4.8",
    "string": "3.x",
    "string-similarity": "^1.2.0",
    "strip-ansi": "^4.0.0",
    "supports-color": "^4.4.0"
  }
}
