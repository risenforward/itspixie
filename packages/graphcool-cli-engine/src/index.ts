export { EnvDoesntExistError } from './errors/EnvDoesntExistError'

export { Config } from './Config'
export { Command } from './Command'
export { CLI, run } from './CLI'
export { Output } from './Output'
export { ProjectDefinitionClass } from './ProjectDefinition/ProjectDefinition'
export {
  RunOptions,
  ProjectDefinition,
  ProjectInfo,
  GraphcoolModule,
  EnvironmentConfig,
  Environments,
  RemoteProject,
  APIError,
  AuthServer,
  AuthTrigger,
  CheckAuth,
  MigrateProjectPayload,
  MigrationActionType,
  MigrationErrorMessage,
  MigrationResult,
  Project,
  Region,
  SchemaInfo,
} from './types'

export { flags, Flag, Arg, Flags } from './Flags'

