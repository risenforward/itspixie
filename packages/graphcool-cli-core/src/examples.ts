/* tslint:disable */
import { ProjectDefinition } from 'graphcool-cli-engine'

export const defaultDefinition: ProjectDefinition = {
  modules: [
    {
      name: '',
      content: `\
# Welcome to Graphcool!
#
# This file is the main config file for your Graphcool Project.
# It's very minimal at this point and uses default values.
# We've included a hello world function here.
# Just uncomment it an run \`graphcool deploy\`
#
# For full config options, check the docs:
#    graph.cool/docs
#
# Happy Coding!


# GraphQL types
types: ./types.graphql


# uncomment this:

# functions:
#   hello:
#     handler:
#       code:
#         src: ./code/hello.js
#     type: resolver
#     schema: ./schemas/hello.graphql

 
# Graphcool modules
modules: {}


# Model/Relation permissions
permissions:
- operation: "*"

  
# Permanent Auth Token / Root Tokens
rootTokens: []

`,
      files: {
        './types.graphql': `\
# This file contains the GraphQL Types
# Graphcool has one special type, the File type:

# type File implements Node {
#   contentType: String!
#   createdAt: DateTime!
#   id: ID! @isUnique
#   name: String!
#   secret: String! @isUnique
#   size: Int!
#   updatedAt: DateTime!
#   url: String! @isUnique
# }

# All types need to have an id field defined like this:

# type User implements Node {
#   id: ID! @isUnique
# }

`,
        './code/hello.js': `\
module.exports = event => {
  return {
    data: {
      message: \`Hello $\{event.data.name || 'World'\}\`
    }
  }
}`,
        './schemas/hello.graphql': `\
type HelloPayload {
  message: String!
}

extend type Query {
  hello(name: String): HelloPayload
}
`,
      },
    },
  ],
}

export const changedDefaultDefinition: ProjectDefinition = {
  modules: [
    {
      name: '',
      content: `
types: ./types.graphql
functions: {}
permissions:
- isEnabled: true
  operation: File.read
  authenticated: false
- isEnabled: true
  operation: File.create
  authenticated: false
- isEnabled: true
  operation: File.update
  authenticated: false
- isEnabled: true
  operation: File.delete
  authenticated: false
- isEnabled: true
  operation: User.read
  authenticated: false
- isEnabled: true
  operation: User.create
  authenticated: false
- isEnabled: true
  operation: User.update
  authenticated: false
- isEnabled: true
  operation: User.delete
  authenticated: false
rootTokens: []
`,
      files: {
        './types.graphql': `type File implements Node {
  contentType: String!
  createdAt: DateTime!
  id: ID! @isUnique
  name: String!
  secret: String! @isUnique
  size: Int!
  updatedAt: DateTime!
  url: String! @isUnique
}

type User implements Node {
  createdAt: DateTime!
  id: ID! @isUnique
  updatedAt: DateTime!
}

type Post {
  imageUrl: String!
}
`,
      },
    },
  ],
}

export const advancedDefinition: ProjectDefinition = {
  modules: [
    {
      name: '',
      content:
        'types: ./types.graphql\nfunctions:\n  filter-posts:\n    handler:\n      code:\n        src: ./code/filter-posts.js\n    type: operationBefore\n    operation: Post.create\n  log-posts:\n    handler:\n      code:\n        src: ./code/log-posts.js\n    type: subscription\n    query: ./code/log-posts.graphql\n  weather:\n    handler:\n      code:\n        src: ./code/weather.js\n    type: resolver\n    schema: ./code/weather.graphql\npermissions:\n- operation: File.read\n- operation: File.create\n- operation: File.update\n- operation: File.delete\n- operation: Post.read\n- operation: Post.update\n- operation: Post.delete\n- operation: Post.create\n  authenticated: true\n- operation: User.read\n- operation: User.create\n- operation: User.update\n- operation: User.delete\nrootTokens: []\n',
      files: {
        './types.graphql':
          'type File implements Node {\n  contentType: String!\n  createdAt: DateTime!\n  id: ID! @isUnique\n  name: String!\n  secret: String! @isUnique\n  size: Int!\n  updatedAt: DateTime!\n  url: String! @isUnique\n}\n\ntype User implements Node {\n  createdAt: DateTime!\n  id: ID! @isUnique\n  updatedAt: DateTime!\n}\n\ntype Post implements Node {\n  title: String!\n  description: String!\n  createdAt: DateTime!\n  id: ID! @isUnique\n  updatedAt: DateTime!\n}',
        './code/log-posts.js':
          '// Click "EXAMPLE EVENT" to see whats in `event`\nmodule.exports = function (event) {\n  console.log(event.data)\n  return {data: event.data}\n}\n',
        './code/log-posts.graphql':
          'subscription {\n  Post(filter: {\n    mutation_in: [CREATED, UPDATED, DELETED]\n  }) {\n    updatedFields\n    node {\n      id\n    }\n  }\n}',
        './code/filter-posts.js':
          "// Click \"EXAMPLE EVENT\" to see whats in `event`\nmodule.exports = function (event) {\n  console.log(event.data)\n  if (event.data.createPost.description.includes('bad') {\n  \treturn {error: 'bad is not allowed'}\n  }\n  return {data: event.data}\n}\n",
        './code/weather.js':
          "const fetch = require('node-fetch')\n\nmodule.exports = function (event) {\n  const city = event.data.city\n  return fetch(getApiUrl(city))\n  .then(res => res.json())\n  .then(data => {\n    console.log(data)\n    return {\n      data: {\n        temperature: data.main.temp,\n        pressure: data.main.pressure,\n        windSpeed: data.wind.speed,\n      }\n    }\n  })\n}\n\nfunction getApiUrl(query) {\n\treturn `http://samples.openweathermap.org/data/2.5/weather?q=${query}&appid=b1b15e88fa797225412429c1c50c122a1`\n  }",
        './code/weather.graphql':
          'type WeatherPayload {\n  temperature: Float!\n  pressure: Float!\n  windSpeed: Float!\n}\n\nextend type Query {\n  weather(city: String!): WeatherPayload\n}\n',
      },
    },
  ],
}

export const examples = {
  blank: defaultDefinition,
  instagram: advancedDefinition,
  stripe: advancedDefinition,
  sendgrid: advancedDefinition,
}
