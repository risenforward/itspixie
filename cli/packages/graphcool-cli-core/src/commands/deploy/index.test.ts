import * as nock from 'nock'
import Deploy from './'
import { changedDefaultDefinition } from '../../examples'
import { mockEnv } from 'graphcool-cli-engine'

afterAll(() => {
  nock.cleanAll()
})

describe('deploy', () => {
  test('without change', async () => {
    const scope = nock('https://api.graph.cool:443', {
      encodedQueryParams: true,
    })
      .post('/system', {
        query:
          '{\n      viewer {\n        user {\n          id\n          email\n        }\n      }\n    }',
      })
      .reply(
        200,
        {
          data: {
            viewer: {
              user: {
                id: 'cj84c8sxb14ze0180pxb30q9r',
                email: 'cli-tests@graph.cool',
              },
            },
          },
        },
        [
          'Content-Type',
          'application/json',
          'Content-Length',
          '94',
          'Connection',
          'close',
          'Date',
          'Thu, 28 Sep 2017 11:41:06 GMT',
          'Request-Id',
          'eu-west-1:system:cj84e2d9s19i90120hvkd39lw',
          'Server',
          'akka-http/10.0.8',
          'X-Cache',
          'Miss from cloudfront',
          'Via',
          '1.1 cdefd7498c238ce7b5d291eb3d7f28f9.cloudfront.net (CloudFront)',
          'X-Amz-Cf-Id',
          'ojrrq0aFbvEOMzgJ63HFD3L6HQ22Ir8HzHYh4LVHUi7VkTKAHWRTxg==',
        ],
      )

    scope
      .post('/system', {
        query:
          '      mutation($projectId: String!, $force: Boolean, $isDryRun: Boolean!, $config: String!) {\n        push(input: {\n          projectId: $projectId\n          force: $force\n          isDryRun: $isDryRun\n          config: $config\n          version: 1\n        }) {\n          migrationMessages {\n            type\n            action\n            name\n            description\n            subDescriptions {\n              type\n              action\n              name\n              description\n            }\n          }\n          errors {\n            description\n            type\n            field\n          }\n          project {\n            id\n            name\n            alias\n            projectDefinitionWithFileContent\n          }\n        }\n      }\n    ',
        variables: {
          projectId: 'cj84dopd3197p01200l5sb9fs',
          isDryRun: false,
          config:
            '{"modules":[{"name":"","content":"# Welcome to Graphcool!\\n#\\n# This file is the main config file for your Graphcool Project.\\n# It\'s very minimal at this point and uses default values.\\n# We\'ve included a hello world function here.\\n# Just uncomment it and run `graphcool deploy`\\n#\\n# Check out some examples:\\n#    github.com/graphcool/examples\\n#\\n# Happy Coding!\\n\\n\\n# GraphQL types\\ntypes: ./types.graphql\\n\\n\\n# uncomment this:\\n\\n# functions:\\n#   hello:\\n#     handler:\\n#       code:\\n#         src: ./code/hello.js\\n#     type: resolver\\n#     schema: ./code/hello.graphql\\n\\n \\n# Graphcool modules\\nmodules: {}\\n\\n\\n# Model/Relation permissions\\npermissions:\\n- operation: \\"*\\"\\n\\n  \\n# Permanent Auth Token / Root Tokens\\nrootTokens: []\\n\\n","files":{"./types.graphql":"# This file contains the GraphQL Types\\n\\n# All types need to have the three fields id, updatedAt and createdAt like this:\\n\\ntype User implements Node {\\n  id: ID! @isUnique\\n  createdAt: DateTime!\\n  updatedAt: DateTime!\\n}\\n\\n\\n# Graphcool has one special type, the File type:\\n\\n# type File implements Node {\\n#   contentType: String!\\n#   createdAt: DateTime!\\n#   id: ID! @isUnique\\n#   name: String!\\n#   secret: String! @isUnique\\n#   size: Int!\\n#   updatedAt: DateTime!\\n#   url: String! @isUnique\\n# }\\n","./code/hello.js":"module.exports = event => {\\n  return {\\n    data: {\\n      message: `Hello ${event.data.name || \'World\'}`\\n    }\\n  }\\n}","./code/hello.graphql":"type HelloPayload {\\n  message: String!\\n}\\n\\nextend type Query {\\n  hello(name: String): HelloPayload\\n}\\n"}}]}',
        },
      })
      .reply(
        200,
        {
          data: {
            push: {
              migrationMessages: [],
              errors: [],
              project: {
                id: 'cj84dopd3197p01200l5sb9fs',
                name: 'Shellcrest Cub',
                alias: null,
                projectDefinitionWithFileContent:
                  '{\n  "modules": [{\n    "name": "",\n    "content": "types: ./types.graphql\\nfunctions: {}\\npermissions: []\\nrootTokens: []\\n",\n    "files": {\n      "./types.graphql": "type User implements Node {\\n  id: ID! @isUnique\\n  createdAt: DateTime!\\n  updatedAt: DateTime!\\n}"\n    }\n  }]\n}',
              },
            },
          },
        },
        [
          'Content-Type',
          'application/json',
          'Content-Length',
          '485',
          'Connection',
          'close',
          'Date',
          'Thu, 28 Sep 2017 11:41:06 GMT',
          'Request-Id',
          'eu-west-1:system:cj84e2de5029j0154vmf95046',
          'Server',
          'akka-http/10.0.8',
          'X-Cache',
          'Miss from cloudfront',
          'Via',
          '1.1 a6cbd015f961a0db7be2b1cd285e26d9.cloudfront.net (CloudFront)',
          'X-Amz-Cf-Id',
          '2L-mJfqGK0CX5ct2XRJ-J_4K1pYMb8Z3CZlWfhBrN6BofdeG5P5e7Q==',
        ],
      )

    const result = await Deploy.mock({mockEnv})
    expect(result.out.stdout.output).toMatchSnapshot()
    scope.done()
  })

  test('with change', async () => {
    nock('https://api.graph.cool:443', { encodedQueryParams: true })
      .post('/system', {
        query:
          '{\n      viewer {\n        user {\n          id\n          email\n        }\n      }\n    }',
      })
      .reply(
        200,
        {
          data: {
            viewer: {
              user: {
                id: 'cj84c8sxb14ze0180pxb30q9r',
                email: 'cli-tests@graph.cool',
              },
            },
          },
        },
        [
          'Content-Type',
          'application/json',
          'Content-Length',
          '94',
          'Connection',
          'close',
          'Date',
          'Thu, 28 Sep 2017 11:47:33 GMT',
          'Request-Id',
          'eu-west-1:system:cj84eao900rw00116ett3sluc',
          'Server',
          'akka-http/10.0.8',
          'X-Cache',
          'Miss from cloudfront',
          'Via',
          '1.1 3f146fa6bc6607097fc0d9bc7e6d4947.cloudfront.net (CloudFront)',
          'X-Amz-Cf-Id',
          '_0Q6Je6TliSoF3mVWkdZNz-k5QvBPPZR0bga4kn0yYYTVGXH9SDXVg==',
        ],
      )

    nock('https://api.graph.cool:443', { encodedQueryParams: true })
      .post('/system', {
        query:
          '      mutation($projectId: String!, $force: Boolean, $isDryRun: Boolean!, $config: String!) {\n        push(input: {\n          projectId: $projectId\n          force: $force\n          isDryRun: $isDryRun\n          config: $config\n          version: 1\n        }) {\n          migrationMessages {\n            type\n            action\n            name\n            description\n            subDescriptions {\n              type\n              action\n              name\n              description\n            }\n          }\n          errors {\n            description\n            type\n            field\n          }\n          project {\n            id\n            name\n            alias\n            projectDefinitionWithFileContent\n          }\n        }\n      }\n    ',
        variables: {
          projectId: 'cj84dopd3197p01200l5sb9fs',
          isDryRun: false,
          config:
            '{"modules":[{"name":"","content":"# This is the changed default definition, used in tests\\n#\\n# This file is the main config file for your Graphcool Project.\\n# It\'s very minimal at this point and uses default values.\\n# We\'ve included a hello world function here.\\n# Just uncomment it and run `graphcool deploy`\\n#\\n# Check out some examples:\\n#    github.com/graphcool/examples\\n#\\n# Happy Coding!\\n\\n\\n# GraphQL types\\ntypes: ./types.graphql\\n\\n\\n# uncomment this:\\n\\n# functions:\\n#   hello:\\n#     handler:\\n#       code:\\n#         src: ./code/hello.js\\n#     type: resolver\\n#     schema: ./code/hello.graphql\\n\\n \\n# Graphcool modules\\nmodules: {}\\n\\n\\n# Model/Relation permissions\\npermissions:\\n- operation: \\"*\\"\\n\\n  \\n# Permanent Auth Token / Root Tokens\\nrootTokens: []\\n\\n","files":{"./types.graphql":"# This file contains the GraphQL Types\\n\\n# All types need to have the three fields id, updatedAt and createdAt like this:\\n\\ntype User implements Node {\\n  id: ID! @isUnique\\n  createdAt: DateTime!\\n  updatedAt: DateTime!\\n}\\n\\n\\n# Graphcool has one special type, the File type:\\n\\n# type File implements Node {\\n#   contentType: String!\\n#   createdAt: DateTime!\\n#   id: ID! @isUnique\\n#   name: String!\\n#   secret: String! @isUnique\\n#   size: Int!\\n#   updatedAt: DateTime!\\n#   url: String! @isUnique\\n# }\\n\\ntype Post implements Node {\\n  id: ID! @isUnique\\n  title: String\\n}\\n","./code/hello.js":"module.exports = event => {\\n  return {\\n    data: {\\n      message: `Hello ${event.data.name || \'World\'}`\\n    }\\n  }\\n}","./code/hello.graphql":"type HelloPayload {\\n  message: String!\\n}\\n\\nextend type Query {\\n  hello(name: String): HelloPayload\\n}\\n"}}]}',
        },
      })
      .reply(
        200,
        {
          data: {
            push: {
              migrationMessages: [
                {
                  name: 'Post',
                  description: 'A new type with the name `Post` is created.',
                  subDescriptions: [
                    {
                      type: 'Field',
                      action: 'Create',
                      name: 'title',
                      description:
                        'A new field with the name `title` and type `String` is created.',
                    },
                  ],
                  type: 'Type',
                  action: 'Create',
                },
              ],
              errors: [],
              project: {
                id: 'cj84dopd3197p01200l5sb9fs',
                name: 'Shellcrest Cub',
                alias: null,
                projectDefinitionWithFileContent:
                  '{\n  "modules": [{\n    "name": "",\n    "content": "types: ./types.graphql\\nfunctions: {}\\npermissions: []\\nrootTokens: []\\n",\n    "files": {\n      "./types.graphql": "type Post implements Node {\\n  id: ID! @isUnique\\n  title: String\\n}\\n\\ntype User implements Node {\\n  id: ID! @isUnique\\n  createdAt: DateTime!\\n  updatedAt: DateTime!\\n}"\n    }\n  }]\n}',
              },
            },
          },
        },
        [
          'Content-Type',
          'application/json',
          'Content-Length',
          '819',
          'Connection',
          'close',
          'Date',
          'Thu, 28 Sep 2017 11:47:34 GMT',
          'Request-Id',
          'eu-west-1:system:cj84eaodb19n701208kfypei3',
          'Server',
          'akka-http/10.0.8',
          'X-Cache',
          'Miss from cloudfront',
          'Via',
          '1.1 6eea7c9b83576b73ff12f8e9ff2ad318.cloudfront.net (CloudFront)',
          'X-Amz-Cf-Id',
          'M-FTdUKUGTug7DBoOpGp-Qr59RvgmPkl8n_FP6NxMRs0VbWhfPE5qQ==',
        ],
      )

    const result = await Deploy.mock({
      mockDefinition: changedDefaultDefinition,
      mockEnv,
    })
    expect(result.out.stdout.output).toMatchSnapshot()
  })
})
