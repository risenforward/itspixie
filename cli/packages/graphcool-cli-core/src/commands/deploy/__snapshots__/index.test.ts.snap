// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deploy with change 1`] = `
"
Your project cj84dopd3197p01200l5sb9fs of env dev was successfully updated.
Here are the changes:


Types

  Post
   + A new type with the name \`Post\` is created.
   └── +  A new field with the name \`title\` and type \`String\` is created.

"
`;

exports[`deploy without change 1`] = `
"Identical project definition for project cj84dopd3197p01200l5sb9fs in env dev, no action required.

"
`;
